pkg/state/state.go

```diff
@@ -334,6 +334,37 @@ func (s *State) PreviousVersion(ctx context.Context, schema string) (*string, er
 	return parent, nil
 }
 
+// Status returns the current migration status of the specified schema
+func (s *State) Status(ctx context.Context, schema string) (*Status, error) {
+	latestVersion, err := s.LatestVersion(ctx, schema)
+	if err != nil {
+		return nil, err
+	}
+	if latestVersion == nil {
+		latestVersion = new(string)
+	}
+
+	isActive, err := s.IsActiveMigrationPeriod(ctx, schema)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
not needed for this PR but we may want to retrieve these two with a single query in the future to guarantee consistency